cmd/stack.go

```diff
@@ -1,38 +1,191 @@
+					planFlag(),
+					regionFlag(),
+				},
+				Action: stackAddCmd,
+			},
+			// {
+			// 	Name:      "remove",
+			// 	ArgsUsage: "[resource-name]",
+			// 	Usage:     "Remove a resource definition from the stack.yml",
+			// 	Action: stack.RemoveResourceCMD
+			// },
+			// {
+			// 	Name:      "plan",
+			// 	Usage:     "Performs a dry run of the stack.yml, showing you the changes that will be made from an apply",
+			// 	Action: stack.PlanCMD,
+			// },
+			// {
+			// 	Name:  "apply",
+			// 	Usage: "List members of a team",
+			// 	Flags: append(teamFlags, []cli.Flag{
+			// 		projectFlag(),
+			// 		cli.BoolFlag{
+			// 			Name:   "upgrade, u",
+			// 			Usage:  "Upgrade any already provisioned products to plans set in stack.yml if upgradable",
+			// 		},
+			// 		cli.BoolFlag{
+			// 			Name:   "interactive, i",
+			// 			Usage:  "Prompt to confirm the project name and prompt to select a plan at every new resource creation, " +
+			// 				"does a dry run and prompts for confirmation before actually running",
+			// 			EnvVar: "MANIFOLD_INTERACTIVE",
+			// 		},
+			// 	}),
+			// 	Action: stack.ApplyCMD,
+			// },
 		},
 	}
 
 	cmds = append(cmds, stackCmd)
 }
 
+func stackAddCmd(cliCtx *cli.Context) error {
+	ctx := context.Background()
+
+	if err := maxOptionalArgsLength(cliCtx, 1); err != nil {
+		return err
+	}
+
+	planName, err := validateName(cliCtx, "plan")
+	if err != nil {
+		return err
+	}
+
+	productName, err := validateName(cliCtx, "product")
+	if err != nil {
+		return err
+	}
+
+	regionName, err := validateName(cliCtx, "region")
+	if err != nil {
+		return err
+	}
+
+	client, err := api.New(api.Catalog)
+	if err != nil {
+		return err
+	}
+
+	catalog, err := catalog.New(ctx, client.Catalog)
+	if err != nil {
+		return cli.NewExitError("Failed to fetch catalog data: "+err.Error(), -1)
+	}
+
+	var plan *cModels.Plan
+
+	products := catalog.Products()
+	productIdx, _, err := prompts.SelectProduct(products, productName)
+	if err != nil {
+		return prompts.HandleSelectError(err, "Could not select product.")
+	}
+
+	if planName != "" {
+		plan, err = catalog.FetchPlanByLabel(ctx, products[productIdx].ID, planName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Plan does not exist.")
+		}
+		_, _, err = prompts.SelectPlan([]*cModels.Plan{plan}, planName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Could not select unlisted plan.")
+		}
+	} else {
+		plans := filterPlansByProductID(catalog.Plans(), products[productIdx].ID)
+		planIdx, _, err := prompts.SelectPlan(plans, planName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Plan does not exist.")
+		}
+		plan = plans[planIdx]
+	}
+
+	regions := filterRegionsForPlan(catalog.Regions(), plan.Body.Regions)
+	_, _, err = prompts.SelectRegion(regions)
+	if err != nil {
+		return prompts.HandleSelectError(err, "Region does not exist.")
+	}
+
+	resourceName, resourceTitle, err := promptNameAndTitle(cliCtx, nil, "resource", true, false)
```

Comment:
Should you use the default from the optional arg here?

Reply:
:+1:

Reply:
its now allows for the title flag
and that gets used implicitly by the name and title func